spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -81,14 +81,16 @@ def get_dynamic_table_property(dynamic_table, *, row_ids=None, property_name):
 
 
 def get_nspikes(units_table, unit_id):
-    """Returns the number of spikes for chosen unit."""
+    """Return the number of spikes for chosen unit."""
     check_nwb_install()
-    if unit_id not in units_table.id[:]:
-        raise ValueError(str(unit_id) + " is an invalid unit_id. "
-                                        "Valid ids: " + str(units_table.id[:].tolist()))
-    nSpikes = np.diff([0] + list(units_table['spike_times_index'].data[:])).tolist()
-    ind = np.where(np.array(units_table.id[:]) == unit_id)[0][0]
-    return nSpikes[ind]
+    ids = np.array(units_table.id[:])
+    indexes = np.where(ids == unit_id)[0]
+    assert len(indexes), ValueError(f"{unit_id} is an invalid unit_id. Valid ids: {ids}.")
```

Comment:
```suggestion
    assert len(indexes), ValueError(f"{unit_id} is an invalid unit_id. Valid ids: {ids}.")
```

Perhaps it is specific to Windows, or my IDE, but whenever I write assertions of the form `assert condition, TypeError(message)`, the final returned error is the `AssertionError`, not the desired `TypeError`. This would be relevant if anything tried to rely on this this function in a `try: except ValueError:` manner. To actually have it return the `ValueError`, this would have to be

```
if not len(indexes):
    raise ValueError(f"{unit_id} is an invalid unit_id. Valid ids: {ids}.")
```

Reply:
@CodyCBakerPhD you're right, the correct syntax is:

```python
assert 1==2, 'error message'
```

if you need another error type, it's better to keep it as:

```python
if error_condition:
    raise ValueError('error message')
```